contracts/cw-proposal-multiple/src/proposal.rs

```diff
@@ -0,0 +1,177 @@
+                let winning_choice_count = self.votes.vote_weights[choice_idx].u128();
+                let remaining_power = self.total_power.sub(self.votes.total()).u128();
+                if winning_choice_count - none_count > remaining_power {
+                    return true;
+                }
+            }
+        }
+
+        false
+    }
+
+    pub fn is_rejected(&self, block: &BlockInfo) -> bool {
+        match (
+            does_vote_count_pass(
+                self.votes.total(),
+                self.total_power,
+                self.voting_strategy.get_quorum(),
+            ),
+            self.expiration.is_expired(block),
+        ) {
+            // Quorum is met and proposal is expired.
+            (true, true) => {
+                // Proposal is rejected if "None" is the winning option.
+                let (_, winning_choice) = self.calculate_winning_choice();
+                if winning_choice.option_type == MultipleChoiceOptionType::None {
+                    return true;
+                }
+
+                // Proposal is rejected if all of the options are tied.
+                if let Some(first) = self.votes.vote_weights.first() {
+                    return self.votes.vote_weights.iter().all(|x| *x == *first);
+                }
+                false
+            }
+            // Quorum is met and proposal has not expired OR Quorum is not met and proposal is not expired.
+            (true, false) | (false, false) => {
+                // Proposal is rejected if "None" has the majority of the total power because there is no way for
+                // another option to outnumber it.
+                let none_vote_count = Uint128::from(self.get_none_vote_count());
+                does_vote_count_fail(
+                    none_vote_count,
+                    self.total_power,
+                    PercentageThreshold::Majority {},
+                )
+            }
+            // Quorum is not met and proposal is expired.
+            (false, true) => true,
+        }
+    }
+
+    pub fn calculate_winning_choice(&self) -> (usize, &MultipleChoiceOption) {
+        match self.voting_strategy {
+            VotingStrategy::SingleChoice { quorum: _ } => {
+                let choice_idx = self
+                    .votes
+                    .vote_weights
+                    .iter()
+                    .enumerate()
+                    .max_by(|a, b| a.cmp(b))
+                    .map(|(idx, _)| idx)
+                    .unwrap();
+
+                (choice_idx, &self.choices[choice_idx])
+            }
+            VotingStrategy::RankedChoice { quorum: _ } => todo!(),
+        }
+    }
+
+    pub fn get_none_vote_count(&self) -> u64 {
+        return self
+            .votes
+            .vote_weights
+            .iter()
+            .enumerate()
+            .filter(|(idx, _x)| self.choices[*idx].option_type == MultipleChoiceOptionType::None)
+            .count() as u64;
```

Comment:
This is the number of folks who have voted for `None` but not the number of votes for `None` as it doesn't count weights here. Think this should sum the weights of votes for `None`.